internal/cadence/lint.go

```diff
@@ -0,0 +1,214 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cadence
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/logrusorgru/aurora/v4"
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/cadence/tools/analysis"
+	"github.com/onflow/flow-cli/internal/command"
+	"github.com/onflow/flowkit/v2"
+	"github.com/onflow/flowkit/v2/output"
+)
+
+type lintFlagsCollection struct{}
+
+type lintResult struct {
```

Comment:
may i suggest we call this something like `lintReport`? so it'll look like
```suggestion
type lintReport struct {
...
...

type lintResults struct {
    Results []lintReport
    ExitCode int

}
```